ql/src/java/org/apache/hadoop/hive/ql/exec/tez/monitoring/RenderStrategy.java

```diff
@@ -39,7 +40,16 @@
   }
 
   private abstract static class BaseUpdateFunction implements UpdateFunction {
-    private static final int PRINT_INTERVAL = 3000;
+    private static final long PRINT_INTERVAL;
+
+    static {
```

Comment:
I think instead of initializing this from a vanilla `new HiveConf` in a static block - we should access this setting from the constructor thru the `monitor` which seem to have a `hiveconf` field

cc: @abstractdog

Reply:
Thanks for the feedback @kgyrtkirk, I've added a `getHiveConf` function to `TezJobMonitor` and move this logic to the constructor of `BaseUpdateFunction`. I guess another approach could be to get `HiveConf` from `SessionState` which is used in the constructor for `perfLogger` already, but grepping for other cases of this approach it doesn't seem to common.

Reply:
This approach seems to be great!
Yes you could get it from SS as well ; but in some cases there might be no seesion in some cases - which should be handled differently.
Thank you!

Reply:
Thanks for the feedback 🙇